.github/workflows/e2e.yaml

```
@@ -97,10 +101,20 @@ jobs:
       - id: generate-cluster-name
         name: generate cluster name
         run: |
-          CLUSTER_NAME="$(echo ${{ inputs.suite }}-"$RANDOM$RANDOM" | awk '{print tolower($0)}' | tr / -)"
-          echo Using cluster name "$CLUSTER_NAME"
+          CLUSTER_NAME=${{ inputs.preexisiting_cluster }}
+          if [[ ${{ inputs.preexisiting_cluster }} == '' ]]; then 
+            if [[ ${{ inputs.workflow_trigger }} == 'sock' ]]; then 
+              CLUSTER_NAME=$(echo soak-periodic-$RANDOM$RANDOM | awk '{print tolower($0)}' | tr / -)
+            else 
+              CLUSTER_NAME=$(echo ${{ inputs.suite }}-$RANDOM$RANDOM | awk '{print tolower($0)}' | tr / -)
+            fi 
+          else 
+            CLUSTER_NAME=${{ inputs.preexisiting_cluster }}
+          fi
+          echo "Using cluster name \"$CLUSTER_NAME\""
           echo CLUSTER_NAME="$CLUSTER_NAME" >> "$GITHUB_OUTPUT"
       - name: create eks cluster '${{ steps.generate-cluster-name.outputs.CLUSTER_NAME }}'
+        if: always() && steps.generate-cluster-name.outputs.CLUSTER_NAME == ''
         uses: ./.github/actions/e2e/create-cluster
```

```suggestion
        uses: ./.github/actions/e2e/setup-cluster
```

Does this become `setup-cluster` since now we are doing more than just cluster creation?